include/deal.II/base/bounding_box.h

```diff
@@ -250,11 +252,22 @@ class BoundingBox
   BoundingBox<spacedim, Number>
   child(const unsigned int index) const;
 
+  /**
+   * The dimensionality of the box returned by cross_section is constrained to
+   * always be at least one to prevent the instantiation of
+   * <code>BoundingBox&lt;0&gt;</code>.
+   *
+   * @note This has to be explicitly calculated outside the method declaration
+   * in a way that works across multiple compilers (e.g., this check cannot be
+   * put directly in the signature of cross_section when using MSVC).
+   */
+  static constexpr int cross_section_dim = spacedim != 1 ? spacedim - 1 : 1;
```

Comment:
Totally not important (and maybe not necessary to do before the release), but I prefer things on the right hand side of `=` to be parenthesized if they contain `?:` or comparison operators. I think it's hard to break down for the eye what the left and right hand side of the central operator is without parentheses. (Central here: `=`.)

Reply:
Separately: You explain in the note why this has to be done outside the declaration in question, but not why we need the `?:` to begin with (or, more precisely, why we need the `else` case).

Reply:
This is so easy to fix lets just do it now. Fixed.

Reply:
My web browser didn't load your second comment for some reason - let me do that too.

Reply:
> You explain in the note why this has to be done outside the declaration in question, but not why we need the ?: to begin with (or, more precisely, why we need the else case

I think this is addressed by the first sentence, where I stated that we need to prevent instantiation of `BoundingBox<0>`. Is that not sufficient?

Reply:
Oh, I didn't read closely enough. Yes.